app/models/manageiq/providers/hawkular/middleware_manager/middleware_deployment.rb

```diff
@@ -1,4 +1,14 @@
 module ManageIQ::Providers
   class Hawkular::MiddlewareManager::MiddlewareDeployment < MiddlewareDeployment
+    def self.resource_path_for_metrics(item)
+      path = ::Hawkular::Inventory::CanonicalPath.parse(item.ems_ref)
```

Comment:
For this, CanonicalPath should be aware that the path is referring to a subdeployment. That shouldn't be problem. So, sounds good! Took note of this, but placing your suggestion in a github issue or an etherpad note may be better? That way, we could ask for team feedback before moving forward.